### PR TITLE
Add support for strings-endpoints

### PR DESCRIPTION
**Pull request by zwilias**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -64,10 +66,26 @@ __Setup Instructions__
...
 In the column names for the values you can specify the language by adding ```$$LANG$$``` this will be replaced by the actual language. For example: ```ios fallback $$LANG$$``` will become ```ios fallback en```.
 
+### Syncing from an endpoint that returns strings files
+
+```json
+{
+  "sync": {
+    "provider": "url",
+    "config": {
+      "url": "https://some.remote.end/point/{{LANG}}.strings",
```
Comment (daanporon) [inline README.md]: formatting issue

File: lib/providers/stringsUrl.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+        return Promise.map(languages, function (language) {
+            var url = syncConfig.url.replace('{{LANG}}', language);
+
+            console.log("Fetching from ", url);
+            return fetch(url).then(function (response) {
+                return response.text();
+            }).then(function (raw) {
+                // TODO: remove when steven fixes backend
+                return raw.replace(/\n/g, ";\n") + ";"
```
Comment (zwilias) [inline lib/providers/stringsUrl.js]: Still need to remove this, so probably best not to merge until that can happen.

File: lib/sync.js [hunk 3]
```diff
@@ -6,155 +6,96 @@ var _ = require('lodash'),
...
+    Promise = require('bluebird'),
+
+    google = require(path.resolve(__dirname, 'providers', 'googleSpreadsheet')),
+    stringsUrl = require(path.resolve(__dirname, 'providers', 'stringsUrl'));
 
 var doubleEscape = function(string) {
     return string.replace(/[^\\]\\(n|r|t)/g, '\\\\$1');
 };
 
+var createProvider = function (config) {
```
Comment (daanporon) [inline lib/sync.js:19]: you can also make this more generic ... and use the filename as provider name ... then you don't need the switch here

File: lib/sync.js [hunk 4]
```diff
@@ -6,155 +6,96 @@ var _ = require('lodash'),
...
+                var provider = createProvider(config);
+
+                provider.configure(config.sync.config)
+                    .then(function (config) {
+                        return locateStringFiles(srcPath)
+                            .then(function (stringFiles) {
+                                return Object.keys(stringFiles);
+                            })
+                            .then(function (languages) {
+                                return provider.getTranslations(config, languages);
```
Comment (daanporon) [inline lib/sync.js:54]: do we need to give the config again ... the provider should already have it from the configure call? it should be configured with the configuration no?

File: lib/providers/googleSpreadsheet.js [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+        return new Promise(function (resolve, reject) {
+            sheet.useServiceAccountAuth(settings.creds, function (err) {
+                if (err) {
+                    reject(err);
+                } else {
+                    resolve(sheet);
+                }
+            })
+        })
+            .then(function (sheet) {
```
Comment (daanporon) [inline lib/providers/googleSpreadsheet.js]: formatting issue

File: README.md (hunk 1 above)
Comment (zwilias) [reply README.md]: Perhaps $$LANG$$ makes more sense, equivalence with how column-names for spreadsheets work?
Other than that, what's the formatting issue here?

File: lib/sync.js (hunk 4 above)
Comment (zwilias) [reply lib/sync.js:54]: Yeah, though to be honest `configure` currently just verifies and processes the config, but doesn't mutate any state. Perhaps I should rename `.configure`, it kind of implies "I'm stateful and this is how you configure me", but that's not what's going on. How about `.processConfig`?

**Pull request merged**